onnxruntime/core/framework/session_state.cc

```diff
@@ -345,17 +345,21 @@ Status SessionState::GeneratePatternGroupCache(const std::vector<std::reference_
   }
   std::unordered_map<std::string, int64_t> map;
   ORT_RETURN_IF_ERROR(ResolveDimParams(*graph_viewer_, feeds, map));
+  return GenerateActivationMemoryPatterns(output, map, resolved_shapes);
+}
+
+Status SessionState::GenerateActivationMemoryPatterns(MemoryPatternGroup* output,
+                                                      std::unordered_map<std::string, int64_t>& map,
+                                                      std::unordered_map<int, TensorShape>& resolved_shapes) const {
   auto* exe_plan = GetExecutionPlan();
   ORT_ENFORCE(exe_plan);
   OrtValuePatternPlanner mem_planner(*exe_plan);
-  auto& node_index_info = GetNodeIndexInfo();
   for (auto& node_plan : exe_plan->execution_plan) {
-    int node_index = node_index_info.GetNodeOffset(node_plan.node_index);
     auto* node = graph_viewer_->GetNode(node_plan.node_index);
-    int output_start = node_index + static_cast<int>(node->InputDefs().size()) + static_cast<int>(node->ImplicitInputDefs().size());
     //allocate output
     for (int i = 0, end = static_cast<int>(node->OutputDefs().size()); i < end; ++i) {
-      const auto ml_value_idx = node_index_info.GetMLValueIndex(output_start + i);
+      int ml_value_idx;
+      this->ort_value_name_idx_map_.GetIdx(node->OutputDefs()[i]->Name(), ml_value_idx);
```

Comment:
nit: don't need "this->"

Reply:
check return value of GetIdx()

Reply:
actually checking the return value may not be needed here - comparing with kInvalidEntry should be good

Reply:
i saw other logic called GetIdx did check the status, let me do it. :) Thanks for catching this,.